### ISSUE TITLE
For 2D and 3D transforms it is not possible to supply all temporary buffers

### ISSUE THREAD

**Issue by AlexHild** [clMathLibraries/clFFT#153]
It would be great to allow the user to supply all needed temporary buffers for 2D and 3D transforms. Currently internal 1D transforms allocate their only temporary buffers. This is a problem when baking using large number of plans, because these temporary buffers are not deallocated until the plan is destroyed.

For now my workaround is to decompose 2D and 3D FFTs into 1D FFTs myself, but this way I do not profit from optimized transpose kernels...

**Comment by bragadeesh** [comment]
Yes, this is a known issue. It is on the list of items to be improved. Unfortunately, I am not in a position to say when this would get worked on.

**Comment by AlexHild** [comment]
I am just running into this issue again. There could be different solutions to this problem:

1) Allow the user to force deallocation of a plans internal buffers (Does not change current API, but allocation of temporary buffers still can not be done by the user)

2) Extend the API to allow passing multiple temporary buffers.

3) Allow the user to pass a pointer to a callback function delivering temporary buffers.

**Comment by bragadeesh** [comment]
Ok, let me reopen. But not sure when we can get to this one. Also, does releasing plans after its use not an option for you?

**Comment by AlexHild** [comment]
Depending on how long recreating plans takes, releasing plans is an option. Will recreating plans be faster than the first time or is the code compiled again?
